File: ROS_WS/src/control/src/cmd_vel_repeater.py
#! /usr/bin/env python
import rospy
import sys, termios, tty, os, time
from geometry_msgs.msg import Twist

vel_msg = Twist()


def move():
    publisher = rospy.Publisher("/cmd_vel", Twist, queue_size=10)
    publisher.publish(vel_msg)


def callback(data):
    global vel_msg
    vel_msg = data


def start():
    rospy.init_node("heartbeat_cmd_vel_repeater", anonymous=False)
    rospy.Subscriber("vel_state", Twist, callback)
    publish_rate = rospy.Rate(25)

    while not rospy.is_shutdown():
        move()
        publish_rate.sleep()


if __name__ == "__main__":
    try:
        start()
    except rospy.ROSInterruptException:
        pass

File: ROS_WS/src/control/src/cmd_vel_repeater.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"

geometry_msgs::Twist active_vel_twist_msg;

void callback(const geometry_msgs::Twist::ConstPtr &msg);

int main(int argc, char **argv)
{
  ros::init(argc, argv, "heartbeat_cmd_vel_repeater");
  ros::NodeHandle n;

  ros::Publisher cmd_vel_publisher = n.advertise<geometry_msgs::Twist>("/cmd_vel", 10);
  ros::Subscriber vel_state_subscriber = n.subscribe<geometry_msgs::Twist>("/vel_state", 10, callback);

  ros::Rate loop_rate(25);
  ROS_INFO("Cmd vel repeater is active.");

  while (ros::ok())
  {
    cmd_vel_publisher.publish(active_vel_twist_msg);
    ros::spinOnce();
    loop_rate.sleep();
  }

  return 0;
}

/**
 * @brief Update the active velocity message
 * @param msg new velocity as Twist
 */
void callback(const geometry_msgs::Twist::ConstPtr &msg)
{
  active_vel_twist_msg = *msg;
}